benchmarks/bench_isolation_forest.py

```diff
@@ -0,0 +1,101 @@
+"""
+==========================================
+IsolationForest benchmark
+==========================================
+
+A test of IsolationForest on classical anomaly detection datasets.
+
+"""
+print(__doc__)
+
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.ensemble import IsolationForest
+from sklearn.metrics import roc_curve, auc
+from sklearn.datasets import load_kddcup99, load_shuttle, fetch_covtype
+from sklearn.preprocessing import LabelBinarizer
+
+np.random.seed(1)
+
+
+datasets = ['http', 'smtp', 'SA', 'SF', 'shuttle', 'forestcover']
+
+for dat in datasets:
+    # loading and vectorization
+    print('loading data')
+    if dat in ['http', 'smtp', 'SA', 'SF']:
+        dataset = load_kddcup99(subset=dat, shuffle=True)
+        X = dataset.data
+        y = dataset.target
+
+    if dat == 'shuttle':
+        dataset = load_shuttle(shuffle=True)
+        X = dataset.data
+        y = dataset.target
+
+        # we remove data with label 4
+        # normal data are then those of class 1
+        s = (y != 4)
+        X = X[s, :]
+        y = y[s]
+        y = (y != 1).astype(int)
+
+    if dat == 'forestcover':
+        dataset = fetch_covtype(shuffle=True)
+        X = dataset.data
+        y = dataset.target
+        # normal data are those with attribute 2
+        # abnormal those with attribute 4
+        s = (y == 2) + (y == 4)
+        X = X[s, :]
+        y = y[s]
+        y = (y != 2).astype(int)
+
+    print('vectorizing data')
+
+    if dat == 'SF':
+        lb = LabelBinarizer()
+        lb.fit(X[:, 1])
+        x1 = lb.transform(X[:, 1])
+        X = np.c_[X[:, :1], x1, X[:, 2:]]
+        y = (y != 'normal.').astype(int)
+
+    if dat == 'SA':
+        lb = LabelBinarizer()
+        lb.fit(X[:, 1])
+        x1 = lb.transform(X[:, 1])
+        lb.fit(X[:, 2])
+        x2 = lb.transform(X[:, 2])
+        lb.fit(X[:, 3])
+        x3 = lb.transform(X[:, 3])
+        X = np.c_[X[:, :1], x1, x2, x3, X[:, 4:]]
+        y = (y != 'normal.').astype(int)
+
+    if dat == 'http' or dat == 'smtp':
+        y = (y != 'normal.').astype(int)
+
+    n_samples, n_features = np.shape(X)
+    n_samples_train = n_samples // 2
+    n_samples_test = n_samples - n_samples_train
+
+    X = X.astype(float)
+    X_train = X[:n_samples_train, :]
+    X_test = X[n_samples_train:, :]
+    y_train = y[:n_samples_train]
+    y_test = y[n_samples_train:]
+
+    print('IsolationForest processing...')
+    model = IsolationForest(bootstrap=True, n_jobs=-1)
+    model.fit(X_train)
+    scoring = model.predict(X_test)
```

Comment:
`scoring` seems to be the opposite of what we want
Areas are close to zero, but close to one if we change `scoring` into `-scoring`